controllers/dspipeline_controller.go

```diff
@@ -231,9 +236,15 @@ func (r *DSPAReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 		if controllerutil.ContainsFinalizer(dspa, finalizerName) {
 			params.Name = dspa.Name
 			params.Namespace = dspa.Namespace
+			params.DSPONamespace = os.Getenv("DSPO_NAMESPACE")
 			if err := r.cleanUpResources(params); err != nil {
 				return ctrl.Result{}, err
 			}
+
+			if err := r.CleanWebhook(ctx, dspa, params); err != nil {
```

Comment:
Could you rename `CleanWebhook` to something like `cleanUpWebhookIfUnused` or something to imply that it only cleans up the webhook if this is the last DPSA with pipeline storage set to Kubernetes.